pyqldbsamples/model/sample_data.py

```diff
@@ -274,8 +274,7 @@ def get_document_ids(transaction_executor, table_name, field, value):
     """
     query = "SELECT id FROM {} AS t BY id WHERE t.{} = '{}'".format(table_name, field, value)
     cursor = transaction_executor.execute_statement(query)
-    list_of_ids = map(lambda table: table.get('id'), cursor)
-    return list_of_ids
+    return list(map(lambda table: table.get('id'), cursor))
```

Comment:
Why has this been done? Everything else looks good.

Reply:
Map returns an iterator which yields results on demand (its not in memory). 

We have updated the functions to call `driver.execute_lambda()` instead of `transaction_executor.execute_statement()` which means after it is out of the lambda, the stream cursor is closed. We need to wrap it to a list to iterate through the map iterator and return an actual list before the stream cursor is closed.

See usage:
```
def get_document_id_by_gov_id(driver, government_id):
    logger.info("Finding secondary owner's person ID using given government ID: {}.".format(government_id))
    return driver.execute_lambda(lambda executor: get_document_ids(executor, Constants.PERSON_TABLE_NAME, 'GovId',
                                                                   government_id))
```

Reply:
In the Python driver we don't have the convenience of auto-buffering?

Reply:
We do, we get a buffered result when we return from a [lambda](https://github.com/awslabs/amazon-qldb-driver-python/blob/a6aee3a2f15ba4869c2cd5c2b72d03ace9f27c22/pyqldb/session/qldb_session.py#L155).

The `get_document_ids` is a pure function and we must process the results within the function before we return.
```
document_ids = driver.execute_lambda(lambda executor: get_document_ids(executor, Constants.PERSON_TABLE_NAME, 'GovId',
                                                   government_id))